extensions/yaml_support/_parser/base_parser.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABCMeta, abstractmethod
+from typing import TypeVar, List
+
+T = TypeVar("T")
+
+class BaseParser(metaclass=ABCMeta):
+    """
+    An abstraction for parsers
+    """
+
+    @abstractmethod
+    def parse(self, code:str) -> List[T]:
+        pass
```

Comment:
We can use a not implementation error for this function.

Reply:
Done